### ISSUE TITLE
[Docs] AbstractVariationalGP explanation/docs

### ISSUE THREAD

**Issue by cherepanovic** [cornellius-gp/gpytorch#897]
Is there an explanation how to set/extend the parameters of AbstractVariationalGP class for my specific purpose?

I have no idea what grid_bounds are and how to choose all parameter properly.

Thank you in advance!

**Comment by gpleiss** [comment]
Hi @cherepanovic - we are actually in the middle of updating the interface of variational inference for our next release (v0.4).

In the meantime, would you be able to describe what you are trying to build? We have examples for lots of common variational models (VGP, SVGP, SV-DKL, etc...) in our [examples](gpytorch.readthedocs.io)

**Comment by cherepanovic** [comment]
@gpleiss 

>In the meantime, would you be able to describe what you are trying to build? 

I am trying to train transformer models [1] with a GP above as a classification task (transformer -> GP -> loss). I tried out the SV-DKL example and it does not work for it. 

I always get the same value for the whole batch in prediction (test function). Hence the Acc is very poor, even under 20%. 

Could you give me an advice how to handle this problem, task? 

I appreciate your help

[1] - https://github.com/huggingface/transformers

**Comment by gpleiss** [comment]
@cherepanovic just to confirm, by "does not work" you mean that the accuracy does not go down? Is that for the test set only, or is it also for the training set? How long do you train the network for?

Two thoughts come to mind:

1) Try following this example: https://github.com/cornellius-gp/gpytorch/blob/master/examples/05_Scalable_GP_Regression_Multidimensional/SVDKL_Regression_SVGP_CUDA.ipynb which uses SVGP instead of SV-DKL. It might produce different results

2) Try training the transformer model (without a GP) first. Then initialize the network in the DKL model with the pre-trained weights.

Additionally, see [this issue](https://github.com/cornellius-gp/gpytorch/issues/882) which discusses troubleshooting DKL models.

Hope this helps! Closing this issue for now :)
